os/mm/mm_heap/mm_node.h

```diff
@@ -24,14 +24,48 @@
  ****************************************************************************/
 
 #include <assert.h>
+#include <debug.h>
+
+#include <tinyara/mm/mm.h>
 
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
 
+#if defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__)
+#define DEBUGASSERT_MM_FREE_NODE(heap, node)		\
+	do {		\
+		DEBUGASSERT(node);		\
+		if (!((node)->blink)) {		\
+			abort_mode = true; \
```

Comment:
DEBUGPANIC() is a macro for `up_assert` function and up_assert sets `abort_mode` to true internally. Is it necessary?

Reply:
```
#if defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__)
#define mfdbg(format, ...)						\
	do {								\
		if (!IS_SECURE_STATE()) {				\
			if (abort_mode || up_interrupt_context()) {	\
				lldbg(format, ##__VA_ARGS__);		\
			} else {					\
				dbg(format, ##__VA_ARGS__);		\
			}						\
		}							\
	} while (0)
#else
#define mfdbg(format, ...) dbg(format, ##__VA_ARGS__)
#endif

```
https://github.com/Samsung/TizenRT/blob/4cced9a8944514cac4c28a2aab2717138b1e474b/os/include/debug.h#L504

mfdbg operation has dependency with abort_mode.

if after print mfdbg( using dbg), up_assert is called, dbg can't finsh printing.
becase uart driver and lib stdout buffer is stop and print only using lldbg.

So, To complete print just before up_assert, abort mode setting is needed.

Reply:
Actullay, This issue is not solved by setting abort_mode early about loadable user heap.. .
So,  I'm checking to add lib stdout flush at up_assert.c